### ISSUE TITLE
CXX-Qt-lib-extra

### ISSUE THREAD

**Issue by LeonMatthesKDAB** [KDAB/cxx-qt#766]
we have discussed many times already, that we don't want to wrap **everything** in CXX-Qt-lib.
As the maintenance burden of this would be quite high.

However, like with `KDToolBox`, we could have a repository of useful things that people have already done, but may not have the time to maintain.

This would be what `CXX-Qt-lib-extra` is.
A collection of wrapped Qt types that we don't really maintain ourselves, but just add whatever we or others find useful.
The barrier to add types here should be quite low, not every part of the API has to be wrapped, just adding stuff is useful, even if it's not complete.

On a technical level, I propose the following:
- Add a `CXX-Qt-lib-extra` independent repository to `KDABLabs`.
- Have `CXX-Qt-lib-extra` depend on the current **stable release** of CXX-Qt. (we don't want to have to drag every part of CXX-Qt-lib-extra through every internal CXX-Qt change).
- Ideally have CXX-Qt-lib-extra depend on `cxx-qt` for the bridges (currently blocked on #598 
- After we release a new version of CXX-Qt, only migrate what still works, fix up anything that's easy to fix, drop support for everything that no longer works... People that need those wrappers can still look the code up in previous versions. After 1.0, this should not happen anymore.

For a draft guideline of when to add something to CXX-Qt-lib directly vs. just to CXX-Qt-lib-extra:
- Default to adding anything you think **may** be useful to CXX-Qt-lib-extra
- Only add types to CXX-Qt-lib directly, if they're either:
    - Hard to wrap optimally (i.e. should be trivial/templates, etc.)
    - Are often used as parameters/return values within the Qt codebase
- We can then always promote types from `lib-extra` to `lib`, if/when it turns out this type is used regularly. 

Looking forward to hear about your opinions :)

**Comment by LeonMatthesKDAB** [comment]
Also note that we may want to move some existing types **out** of CXX-Qt-lib into extra.
Which remains to be discussed :)

**Comment by Montel** [comment]
qmargins for example ? Not sure that it's used by default no ?

**Comment by ahayzen-kdab** [comment]
Initially I went through the QML basic types https://doc.qt.io/qt-5/qtqml-typesystem-basictypes.html and tried to wrap those as "fundamental types" that we require, note how that has point, rect, size. And then note that QRect has API like QRect::marginsAdded, so margins are a curious one :-)

**Comment by Montel** [comment]
CXX-Qt-lib-extra lives for the moment in official repo.
We will split after 0.7.
I added QCommandLineParser/QCommandOption/QElipsedTimer.

**Comment by Montel** [comment]
I added QApplication in  #869
In review for the moment we must find a solution for implement inherit class

**Comment by Montel** [comment]
Add QStandardPaths ? 
It can be useful.
We can define [setTestModeEnabled](https://doc.qt.io/qt-6/qstandardpaths.html#setTestModeEnabled)

**Comment by LeonMatthesKDAB** [comment]
This is mostly done, it is just not yet moved out of this repository. For now we can keep it in this respository unless the maintenance burden increases or there is another concrete reason to move it.
